grammars/css.cson

```diff
@@ -20,294 +20,1176 @@
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
         'patterns': [
           {
-            'include': '#media-query-list'
+            'include': '#selector-innards'
           }
         ]
       }
       {
-        'begin': '\\s*(\\{)'
+        # Child-indexed
+        'begin': '(?i)((:)nth-(?:last-)?(?:child|of-type))(\\()'
         'beginCaptures':
           '1':
-            'name': 'punctuation.section.property-list.begin.css'
-        'end': '(?=\\})'
+            'name': 'entity.other.attribute-name.pseudo-class.css'
+          '2':
+            'name': 'punctuation.definition.entity.css'
+          '3':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
         'patterns': [
           {
-            'include': '$self'
+            'match': '(?i)[+-]?(\\d+n?|n)(\\s*[+-]\\s*\\d+)?'
+            'name': 'constant.numeric.css'
+          }
+          {
+            'match': '(?i)even|odd'
+            'name': 'support.constant.parity.css'
           }
         ]
       }
     ]
-  }
-  {
-    'begin': '(@)page\\b'
-    'captures':
-      '0':
-        'name': 'keyword.control.at-rule.page.css'
-      '1':
-        'name': 'punctuation.definition.keyword.css'
-    'end': '\\s*((?=[{:]|$))'
-    'name': 'meta.at-rule.page.css'
-    'patterns': [
-      {
-        'include': '#rule-list'
-      }
-    ]
-  }
-  {
-    'begin': '((@)namespace\\b)\\s*(?=url)'
-    'captures':
-      '1':
-        'name': 'keyword.control.at-rule.namespace.css'
-      '2':
-        'name': 'punctuation.definition.keyword.css'
-      '3':
-        'name': 'support.function.misc.css'
-    'comment': 'Namespace without prefix'
-    'end': '\\s*((?=;|$))'
-    'name': 'meta.at-rule.namespace.css'
-    'patterns': [
-      {
-        'include': '#property-values'
-      }
-      {
-        'include': '#string-double'
-      }
-      {
-        'include': '#string-single'
-      }
-    ]
-  }
-  {
-    'begin': '((@)namespace) ([\\w-]*)'
+  'media-features':
     'captures':
       '1':
-        'name': 'keyword.control.at-rule.namespace.css'
+        'name': 'support.type.property-name.media.css'
       '2':
-        'name': 'punctuation.definition.keyword.css'
+        'name': 'invalid.deprecated.media.css'
       '3':
-        'name': 'entity.name.function.css'
-    'comment': 'Namespace'
-    'end': '\\s*((?=;|$))'
-    'name': 'meta.at-rule.namespace.css'
+        'name': 'support.type.vendored.property-name.media.css'
+    'match': '''(?xi)
+      (?<=^|\\s|\\(|\\*/)           # Preceded by whitespace, bracket or comment
+
+      (?:
+        # Standardised features
+        (
+          (?:min-|max-)?
+          (?: height
+            | width
```

Comment:
Think I'll include a sidebar annotation instead, so the correlation [with the spec](https://drafts.csswg.org/mediaqueries/#mq-ranges) is clearer:
<img src="https://cloud.githubusercontent.com/assets/2346707/19753549/46a3b97c-9c51-11e6-8a51-578a0f36bf4b.gif" width="512" alt="Nah" />